app/services/ipam/eui64.rb

```diff
@@ -1,16 +1,27 @@
 module IPAM
   class Eui64 < Base
     def suggest_ip
-      logger.debug("Suggesting ip for #{self} based on mac '#{mac}' (EUI-64).")
-      return unless mac.present? && subnet.network.present? && subnet.cidr.present?
-      raise Foreman::Exception.new(N_("Prefix length must be /64 or less to use EUI-64")) if subnet.cidr > 64
+      logger.debug("Suggesting ip for #{subnet} based on mac '#{mac}' (EUI-64).")
+      validate
+      return if errors.present?
       ip = mac_to_ip(mac)
       logger.debug("Generated #{ip}")
       ip
     end
 
+    def suggest_new?
+      false
+    end
+
     private
 
+    def validate
+      errors.add(:mac, _("can't be blank")) unless mac.present?
+      errors.add(:subnet, _("network can't be blank")) unless subnet.network.present?
+      errors.add(:subnet, _("CIDR can't be blank")) unless subnet.cidr.present?
```

Comment:
"did you mean prefix length?" - ![clippy](https://images.duckduckgo.com/iu/?u=http%3A%2F%2Fcdn.geekwire.com%2Fwp-content%2Fuploads%2F2011%2F04%2Fclippy.jpg&f=1)